cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -117,8 +166,19 @@ def _add_login_nodes_pool_auto_scaling_group(self):
             vpc_zone_identifier=self._pool.networking.subnet_ids,
         )
 
+        self._add_lifecycle_hook(auto_scaling_group)
+
         return auto_scaling_group
 
+    def _add_lifecycle_hook(self, auto_scaling_group):
+        return autoscaling.CfnLifecycleHook(
+            self,
+            "LoginNodesASGLifecycleHook",
```

Comment:
Do we need any additional IAM policies to create/manage a LifeCycleHook resource?

Reply:
I think we need:
autoscaling:PutLifecycleHook -> create the lifecycle hook

Reply:
Thanks, let's add it to the list of permissions needed for the API as well.